extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1102,35 +1146,45 @@ public void run()
                 final Map<String, Set<SegmentIdentifier>> segmentsToMoveOut = new HashMap<>();
 
                 for (InputRow row : rows) {
-                  if (row != null && withinMinMaxRecordTime(row)) {
-                    final String sequenceName = sequenceNames.get(record.partition());
-                    final AppenderatorDriverAddResult addResult = driver.add(
-                        row,
-                        sequenceName,
-                        committerSupplier,
-                        false,
-                        false
-                    );
-
-                    if (addResult.isOk()) {
-                      // If the number of rows in the segment exceeds the threshold after adding a row,
-                      // move the segment out from the active segments of BaseAppenderatorDriver to make a new segment.
-                      if (addResult.getNumRowsInSegment() > tuningConfig.getMaxRowsPerSegment()) {
-                        segmentsToMoveOut.computeIfAbsent(sequenceName, k -> new HashSet<>())
-                                         .add(addResult.getSegmentIdentifier());
+                  try {
+                    if (row != null && withinMinMaxRecordTime(row)) {
+                      final String sequenceName = sequenceNames.get(record.partition());
+                      final AppenderatorDriverAddResult addResult = driver.add(
+                          row,
+                          sequenceName,
+                          committerSupplier,
+                          false,
+                          false
+                      );
+
+                      if (addResult.isOk()) {
+                        // If the number of rows in the segment exceeds the threshold after adding a row,
+                        // move the segment out from the active segments of BaseAppenderatorDriver to make a new segment.
+                        if (addResult.getNumRowsInSegment() > tuningConfig.getMaxRowsPerSegment()) {
+                          segmentsToMoveOut.computeIfAbsent(sequenceName, k -> new HashSet<>())
+                                           .add(addResult.getSegmentIdentifier());
+                        }
+                        isPersistRequired |= addResult.isPersistRequired();
+                      } else {
+                        // Failure to allocate segment puts determinism at risk, bail out to be safe.
+                        // May want configurable behavior here at some point.
+                        // If we allow continuing, then consider blacklisting the interval for a while to avoid constant checks.
+                        throw new ISE("Could not allocate segment for row with timestamp[%s]", row.getTimestamp());
                       }
-                      isPersistRequired |= addResult.isPersistRequired();
+
+                      if (addResult.getParseException() != null) {
+                        throw addResult.getParseException();
```

Comment:
Same here.

Reply:
Changed this and similar places to use handleParseException